File: AEDII-FELIPE-24-1/TP01/Q01eQ02/Palindromo.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

// Função para verificar se a entrada é "FIM"
bool isFim(char *s) {
    return strcmp(s, "FIM") == 0;
}

// Função para verificar se uma string é um palíndromo
bool isPalindromo(char *s) {
    int inicio = 0;
    int fim = strlen(s) - 1;
    
    while (inicio < fim) {
        if (s[inicio] != s[fim]) {
            return false;
        }
        inicio++;
        fim--;
    }
    return true;
}

// Função principal
int main() {
    char entrada[100];

    do {
        fgets(entrada, 100, stdin);
        entrada[strcspn(entrada, "\n")] = '\0'; // Remover a quebra de linha
        
        if (!isFim(entrada)) {
            if (isPalindromo(entrada)) {
                printf("SIM\n");
            } else {
                printf("NAO\n");
            }
        }
    } while (!isFim(entrada));
    
    return 0;
}

File: AED's/AEDII/AED2-FELIPE-24-1/TP01/Q01eQ02/Palindromo.java
package TP01.Q01eQ02;
class Palindromo {
    /**
     *  Verifica se a entrada é igual a "FIM".
     */
    public static boolean isFim(String s) {
        return s.equals("FIM");
    }

    /**
     *  Verifica se a string é um palíndromo.
     */
    public static boolean isPalindromo(String s) {
        int inicio = 0;
        int fim = s.length() - 1;
        
        while (inicio < fim) {
            if (s.charAt(inicio) != s.charAt(fim)) {
                return false;
            }
            inicio++;
            fim--;
        }
        return true;
    }

    /**
     *  Função principal.
     */
    public static void main(String[] args) {
        String entrada = "";

        do {
            entrada = MyIO.readLine();
            if (!isFim(entrada)) {
                if (isPalindromo(entrada)) {
                    MyIO.println("SIM");
                } else {
                    MyIO.println("NAO");
                }
            }
        } while (!isFim(entrada));
    }
}
